### DISCUSSION TITLE
Compatibility inquiry: lmstfy version 1.0.11

### DISCUSSION THREAD

**Discussion by lbydev** [bitleak/lmstfy#217] [Q&A] [answered]
We have been using version 1.0.10 of the lmstfy library (https://github.com/bitleak/lmstfy) in our project for some time. Our server is running version 1.0.10, and we are currently using the client version v1.0.4.

However, we recently noticed that version 1.0.11 has been released. Before upgrading to the latest version, we wanted to inquire about its compatibility with our existing codebase.

If you have already upgraded to version 1.0.11 or have any insights on its compatibility, we would greatly appreciate your input. Specifically, we would like to know if any breaking changes were introduced in this version and if there are any considerations or best practices we should follow during the upgrade process.

Please note that our server version is 1.0.10, and we are using the client version v1.0.4.

Thank you in advance for your help and expertise. We look forward to hearing from you.

Best regards,

**Comment by git-hulk** [comment, marked as answer]
Hi @lbydev,

v1.0.11 has been released for more than 2 years and we've been using it in production for a while. 

> Please note that our server version is 1.0.10, and we are using the client version v1.0.4.

It's good to upgrade the server version to 1.0.10 and keep the client version as it is. You can upgrade lmstfy instances one by one if you worry about this.

**Comment by lbydev** [comment]
Hello,

Thank you for your prompt response. I appreciate your explanation regarding the differences between versions v1.0.10 and v1.0.11 of lmstfy, as well as the refactorings that have taken place, such as the removal of the "freeze tries" parameter in the BatchConsume interface.

Considering the significant number of modified files and the refactorings, it is understandable that there might be compatibility concerns. To address this, I will proceed with upgrading the lmstfy version to v1.0.11 in a test environment. I will also test the compatibility with the client version v1.0.11 to ensure a smooth transition.

I will keep you updated on the results of the regression testing using the upgraded lmstfy version. If any issues or further questions arise during the testing process, I will reach out to you for assistance.

Once again, I appreciate your support.

**Reply by git-hulk** [reply]
Yes, `freeze_tries` is a deprecated feature and didn't export to users at all, so it's good to remove it. But LMSTFY's release notes are not well-maintained, will raise formal release logs at next time.

**Reply by lbydev** [reply]
I would like to express my heartfelt gratitude for your patient and informative response. I appreciate your efforts in addressing my concerns regarding the compatibility issues in LMSTFY and providing clarification on the removal of the deprecated feature, "freeze_tries."

I understand that maintaining release notes can be challenging, and I appreciate your commitment to improving this aspect in future updates. Your dedication to open-source development and the LMSTFY project is truly commendable.

Moving forward, I will proceed with the gradual upgrade and thorough testing of LMSTFY in our testing environment. I will keep you informed of the progress and any feedback or issues that arise during the process.

Once again, thank you for your valuable assistance, open-source spirit, and contributions to the development community. Your support is greatly appreciated.
